src/data/actions/CELLAR_V0815/getApy.ts

```diff
@@ -54,21 +39,24 @@ export const getApy = async (
     const cellarApy = (() => {
       const indexThatHaveChanges = dayDatas.findIndex(
         (data, idx, arr) => {
-          if (idx === 0) return false // return false because first data doesn't have comparation
+          if (idx === 0) return false // return false because first data doesn't have comparison
 
-          const prev = arr[idx + 1].shareValue
+          const prev = arr[idx - 1].shareValue
           return prev !== data.shareValue
         }
       )
       if (indexThatHaveChanges === -1) return 0
+
+      // dayDatas is in desc order
       const latestDataChanged =
-        dayDatas[indexThatHaveChanges].shareValue
+        dayDatas[indexThatHaveChanges - 1].shareValue
```

Comment:
The dayData is returned in descending date order. Therefore, the latest / newer element will always have a lower index.